modules/imgproc/src/pyramids.cpp

```diff
@@ -1380,9 +1380,19 @@ void cv::pyrUp( InputArray _src, OutputArray _dst, const Size& _dsz, int borderT
     CV_OCL_RUN(_src.dims() <= 2 && _dst.isUMat(),
                ocl_pyrUp(_src, _dst, _dsz, borderType))
 
+    // Define maximum allowable image dimensions to prevent memory overflow
```

Comment:
i think you can use `validateInputImageSize`
look
https://github.com/opencv/opencv/blob/4c024c35fbc7f0610501e087a9ef20c336a75e2b/modules/imgcodecs/src/loadsave.cpp#L72-L81